guppylang/cfg/cfg.py

```diff
@@ -67,3 +59,13 @@ def link(self, src_bb: BB, tgt_bb: BB) -> None:
         """Adds a control-flow edge between two basic blocks."""
         src_bb.successors.append(tgt_bb)
         tgt_bb.predecessors.append(src_bb)
+
+    def analyze(
```

Comment:
Not strictly necessary, but I thought it made sense to move it since the analysis pass that is run on the `CheckedCFG` for linearity checking uses different stats